docs/next-env.d.ts

```diff
@@ -1,6 +1,6 @@
 /// <reference types="next" />
 /// <reference types="next/image-types/global" />
-/// <reference path="./.next/types/routes.d.ts" />
+/// <reference path="./export/types/routes.d.ts" />
```

Comment:
@Janpot do you know why we have this file changing on PRs?

Reply:
Yeah, it's `next dev` changing it to 
```tsx
/// <reference path="./.next/types/routes.d.ts" />
```
and `next build` changing it to
```tsx
/// <reference path="./export/types/routes.d.ts" />
```
I think we can consider this a bug in next.js, it hasn't been reported yet, but I wouldn't hold my breath waiting for it to ever be fixed.

We should keep
```tsx
/// <reference path="./.next/types/routes.d.ts" />
```

Reply:
Tackling in https://github.com/mui/mui-x/pull/20177